src/core/atom.ts

```diff
@@ -27,18 +27,30 @@ export class Atom implements IAtom {
     diffValue = 0
     lastAccessedBy = 0
     lowestObserverState = IDerivationState.NOT_TRACKING
+
+    private observedListeners?: Set<Function>
+    private unobservedListeners?: Set<Function>
+
+    get onBecomeObservedListeners(): Set<Function> {
+        if (this.observedListeners === undefined) this.observedListeners = new Set()
```

Comment:
The getter causes the set to be initializaed lazily, but effectively for every atom, correct (as MobX core will often inspect this getter)? So this means that every atom now will have an additional `Set` to allocate. 

I would prefer to see a solution that doesn't allocate the Set (or just a shared, empty, frozen one), until the first listeners is established, and only then instantiate the local Set of listeners